pkg/manifests/manifests.go

```diff
@@ -350,6 +353,14 @@ func (f *Factory) AlertmanagerTrustedCABundle() (*v1.ConfigMap, error) {
 	return cm, nil
 }
 
+const (
+	// These constants refer to indices of alertmanager-main containers.
+	// They need to be in sync with jsonnet/alertmanager.jsonnet
+	ALERTMANAGER_CONTAINER_OAUTH_PROXY      = 0
+	ALERTMANAGER_CONTAINER_KUBE_RBAC_PROXY  = 1
+	ALERTMANAGER_CONTAINER_PROM_LABEL_PROXY = 2
```

Comment:
in other places we instead started converting this to a container name based scheme, to avoid having to guarantee order in generated files, not a blocker, but since this is net new maybe worth doing

Reply:
ok I wasn't sure since I've at least 3 ways of doing it but happy to fix it in this PR!

Reply:
Yes, please, container name based approach is better as we have had few problems with the ordered approach as we do not know if those change, whereas name based will fail, and we should at least be migrating away from this IMHO.